zerver/lib/actions.py

```diff
@@ -4513,6 +4522,33 @@ def do_update_message(user_profile: UserProfile, message: Message,
         event["new_stream_id"] = new_stream.id
         event["propagate_mode"] = propagate_mode
 
+        # When messages are moved from one stream to another, some
+        # users may lose access to those messages, including guest
+        # users and users not subscribed to the new stream (if it is a
+        # private stream).  For those users, their experience is as
+        # though the messages were deleted, and we should send a
+        # delete_message event to them instead.
+
+        subscribers = get_active_subscriptions_for_stream_id(
+            stream_id).select_related("user_profile")
+        subs_to_new_stream = get_active_subscriptions_for_stream_id(
+            new_stream.id).select_related("user_profile")
+
+        # TODO: Should we put a list() around the above queries to
```

Comment:
Can you look at this TODO?

Reply:
I don't think it really makes any difference but I did it for subs_to_new_stream .